soc/nordic/validate_binding_headers.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * This file validates definitions found in dt-bindings headers against their
+ * expected values from MDK, which may be provided in the form of C types.
+ *
+ * Note: all dt-bindings headers which have been included by DTS in this build
+ * are automagically included in this file as well. See CMakeLists.txt.
+ */
+
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/toolchain.h>
+
+#include <nrf.h>
+
+/**
+ * Domain IDs. See:
+ *   - dt-bindings/misc/nordic-domain-id-nrf54h20.h
```

Comment:
note that by doing:

```suggestion
#include <zephyr/dt-bindings/misc/nordic-domain-id-nrf54h20.h>
```

you'd effectively avoid any CMake automagic and file would be self-explanatory ;-)

Reply:
Well, it would start more like this:
```c
#ifdef CONFIG_SOC_NRF54H20
#include <zephyr/dt-bindings/misc/nordic-domain-id-nrf54h20.h>
#endif /* CONFIG_SOC_NRF54H20 */
```
But yeah, the automagic feels less valuable with every force push.